tests/unit/controlgroup/core.js

```diff
@@ -160,4 +160,39 @@ QUnit.test( "Single controlgroup button - vertical", function( assert ) {
 		" ui-corner-tr ui-corner-tl ui-corner-bl ui corner-br" );
 } );
 
+QUnit.module( "Controlgroup: Non-empty class key", {
+	setup: function() {
+		this.classKey = $.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ];
+		$.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] =
+			$.trim( $.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] +
+				" something-custom" );
+	},
+	teardown: function() {
+		$.ui.selectmenu.prototype.options.classes[ "ui-selectmenu-button-closed" ] = this.classKey;
+	}
+} );
+
+QUnit.test( "Controlgroup handles non-empty class key correctly", function( assert ) {
+	assert.expect( 2 );
+
+	$( ".controlgroup-non-empty-class-key" )
+		.controlgroup()
+		.controlgroup( "instance" );
+
+	assert.deepEqual(
```

Comment:
I still dont think we need to test for that when the bug only existed as part of a greater bug that would have also failed with just a `hasClasses` check.

Alternatively if you really think this is important for you to check make it a separate test and just use `matches` on `instance.options.classes[ 'whatever-key' ]` and check the length of the class you added it should never be more then 1.